test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsUtilityTests.cs

```diff
@@ -316,5 +316,89 @@ public void GetDefaultPushSource_WithNullSettings_Throws()
             ex.Should().NotBeNull();
             ex.Should().BeOfType<ArgumentNullException>();
         }
+
+        [Fact]
+        public void GetNamespaceMode_WithNullSettings_Throws()
+        {
+            var ex = Record.Exception(() => SettingsUtility.GetNamespaceMode(settings: null));
+
+            ex.Should().NotBeNull();
+            ex.Should().BeOfType<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void GetNamespaceMode_WithValidKeyValue_Success()
+        {
+            // Arrange
+            var nugetConfigPath = "NuGet.Config";
+            var config = @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <config>
+        <add key=""namespaceMode"" value=""singleSourcePerPackage"" />
```

Comment:
Is it applying same mode for all sources? Don't some users want to set for per source?

Reply:
> Is it applying same mode for all sources?

Yes, Erick that is correct. 

> Don't some users want to set for per namespace?

This scenario is not referred in the [spec](https://github.com/NuGet/Home/blob/dev/proposed/2021/PackageNamespaces.md). Hence, it is out of scope for now.